feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,400 @@
+        # inputs
+        self.variables = _define_variables(variables)
+        self.vars_type = variables_type
+        self.metrics_score = metrics_score
+        self.target = target
+        self.test_size = test_size
+        self.random_state = random_state
+        self.quantiles = quantiles
+
+        # transformed variables
+        self.target_enc_dict = {}
+        self.transformed_cols = []
+        self.binned_vars = []
+        
+        # dataframes
+        self.X_train, self.X_test = None, None
+        self.y_train, self.y_test = None, None
+        self.X_train_enc, self.X_test_enc = None, None
+        
+        # output
+        self.output = None
+
+
+    def __target_mean_encoding(self, df_train_, df_test_, var_cols=None):
+        """
+        Performs
+        --------
+            For each variable:
+                + Performs the target mean encoding on the train set. 
+                + Applies the result to both train and test sets.
+            Drop the target variable from the datasets.
+            Return the updated target mean encoding datasets.
+
+        Returns
+        -------
+        df_train: pandas dataframe of shape = [n_samples, n_features]
+            The output train set dataframe.
+
+        df_test: pandas dataframe of shape = [n_samples, n_features]
+            The output test set dataframe.
+
+        Parameters
+        ----------
+        df_train_: pandas dataframe of shape = [n_samples, n_features]
+            The input train set dataframe.
+
+        df_test_: pandas dataframe of shape = [n_samples, n_features]
+            The input test set dataframe.
+
+        var_cols: list, default=None
+            The list of variables that is needed to be transformed 
+            the target mean encoding.
+
+
+        Attributes
+        ----------
+        __target_mean_enc: dict, default=None
+            The output of group of variables by target mean of the train set.
+
+        target_enc_dict: dict, default=None
+            The dictionary that stores the result of target encoding of 
+            each variable in the train set.
+
+        """
+        df_train, df_test = df_train_.copy(), df_test_.copy()
+        for col in var_cols:
+            # get the target encoding on train set.
+            __target_mean_enc = df_train_.groupby([col])[self.target].mean().to_dict()
+            # update this target encoding result to both train and test sets.
+            df_train[col] = df_train_[col].map(__target_mean_enc)
+            df_test[col] = df_test_[col].map(__target_mean_enc)
+            self.target_enc_dict[col] = __target_mean_enc
+
+        # drop target from the updated datasets
+        df_train.drop([self.target], axis=1, inplace=True)
+        df_test.drop([self.target], axis=1, inplace=True)
+
+        return df_train, df_test 
+
+    def __validate_variables(self, X):
+        """
```

Comment:
we have a lot of code to validate variables in the variable_manipulation.py. Can we use those functions instead?

Reply:
Yes, I will use those functions for validation. Thanks Sole!